packages/SwingSet/test/test-vatstore.js

```diff
@@ -0,0 +1,54 @@
+/* global require */
+// eslint-disable-next-line import/order
+import { test } from '../tools/prepare-test-env-ava';
+
+import { provideHostStorage } from '../src/hostStorage';
+import { buildVatController } from '../src/index';
+import { capargs } from './util.js';
+
+test('vatstore', async t => {
+  const config = {
+    bootstrap: 'bootstrap',
+    vats: {
+      bootstrap: {
+        sourceSpec: require.resolve('./vat-vatstore.js'),
+        creationOptions: {
+          enableVatstore: true,
+        },
+      },
+    },
+  };
+  const hostStorage = provideHostStorage();
+  const c = await buildVatController(config, [], {
+    hostStorage,
+  });
+
+  function send(msg, ...args) {
+    c.queueToVatExport('bootstrap', 'o+0', msg, capargs(args));
+  }
+
+  send('get', 'zot');
+  send('store', 'zot', 'first zot');
+  send('get', 'zot');
+  send('get', 'foo');
+  send('store', 'zot', 'second zot');
+  send('store', 'foo', 'first foo');
+  send('get', 'zot');
+  send('delete', 'zot');
+  send('get', 'foo');
+  send('get', 'zot');
+  await c.run();
+
```

Comment:
Let's add another check here, peek into the kvStore and confirm the surviving `foo` key is being stored with the userspace-specific prefix, not something which could possibly overlap with e.g. the virtual object manager's keyspace.